lib/msf/core/payload/stager.rb

```diff
@@ -156,9 +156,7 @@ def generate_stage(opts={})
   def handle_connection(conn, opts={})
     # If the stage should be sent over the client connection that is
     # established (which is the default), then go ahead and transmit it.
-    if (stage_over_connection?)
-      opts = {}
```

Comment:
I was the culprit that added this line where the opts were completely removed. I am not sure why. There are no comments here saying why, and so I am thinking it was a bug that didn't raise its ugly head until I tried to pass through the `skip_ssl` option.